src/Microsoft.OpenApi/Services/OpenApiWalker.cs

```diff
@@ -787,6 +808,7 @@ internal void Walk(IDictionary<string,OpenApiExample> examples)
             {
                 foreach (var example in examples)
                 {
+                    _visitor.CurrentKeys.Example = example.Key;
                     Walk(example.Key, () => Walk(example.Value));
                 }
             }
```

Comment:
set the CurrentKeys.Example to null after the loop

Reply:
What do you think about putting it inside the loop instead?

                foreach (var example in examples)
                {
                    _visitor.CurrentKeys.Example = example.Key;
                    Walk(example.Key, () => Walk(example.Value));
                    _visitor.CurrentKeys.Example = null;
                }

I know it's less efficient, but it's more semantically correct in my opinion. Once we finish processing that example object, we should set the key back to null, instead of relying on the value being overridden by the next loop item.

---
In reply to: [194930193](https://github.com/Microsoft/OpenAPI.NET/pull/280#discussion_r194930193) [](ancestors = 194930193)

Reply:
I'm ok with that.

---
In reply to: [194930909](https://github.com/Microsoft/OpenAPI.NET/pull/280#discussion_r194930909) [](ancestors = 194930909,194930193)